src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -0,0 +1,214 @@
+    m_scriptWrappedFunctionCache.clear();
+    m_incomingDataFunctions.clear();
+    m_scriptFunctionPrefixes.clear();
+    ControllerScriptEngineBase::shutdown();
+}
+
+bool ControllerScriptEngineLegacy::handleIncomingData(const QByteArray& data) {
+    // This function is called from outside the controller engine, so we can't
+    // use VERIFY_OR_DEBUG_ASSERT here
+    if (!m_pJSEngine) {
+        return false;
+    }
+
+    QJSValueList args;
+    args << m_pJSEngine->toScriptValue(data);
+    args << QJSValue(data.size());
+
+    for (const QJSValue& function : m_incomingDataFunctions) {
+        ControllerScriptEngineBase::executeFunction(function, args);
+    }
+
+    return true;
+}
+
+bool ControllerScriptEngineLegacy::evaluateScriptFile(const QFileInfo& scriptFile) {
+    VERIFY_OR_DEBUG_ASSERT(m_pJSEngine) {
+        return false;
+    }
+
+    if (!scriptFile.exists()) {
+        qWarning() << "ControllerScriptHandlerLegacy: File does not exist:"
+                   << scriptFile.absoluteFilePath();
+        return false;
+    }
+    m_scriptWatcher.addPath(scriptFile.absoluteFilePath());
+
+    qDebug() << "ControllerScriptHandlerLegacy: Loading"
+             << scriptFile.absoluteFilePath();
+
+    // Read in the script file
+    QString filename = scriptFile.absoluteFilePath();
+    QFile input(filename);
+    if (!input.open(QIODevice::ReadOnly)) {
+        qWarning() << QString(
+                "ControllerScriptHandlerLegacy: Problem opening the script file: %1, "
+                "error # %2, %3")
+                              .arg(filename,
+                                      QString::number(input.error()),
+                                      input.errorString());
+        // Set up error dialog
+        ErrorDialogProperties* props = ErrorDialogHandler::instance()->newDialogProperties();
+        props->setType(DLG_WARNING);
+        props->setTitle(tr("Controller Mapping File Problem"));
+        props->setText(tr("The mapping for controller \"%1\" cannot be opened.")
+                               .arg(m_pController->getName()));
+        props->setInfoText(
+                tr("The functionality provided by this controller mapping will "
+                   "be disabled until the issue has been resolved."));
+
+        // We usually don't translate the details field, but the cause of
+        // this problem lies in the user's system (e.g. a permission
+        // issue). Translating this will help users to fix the issue even
+        // when they don't speak english.
+        props->setDetails(tr("File:") + QStringLiteral(" ") + filename +
+                QStringLiteral("\n") + tr("Error:") + QStringLiteral(" ") +
+                input.errorString());
+
+        // Ask above layer to display the dialog & handle user response
+        ErrorDialogHandler::instance()->requestErrorDialog(props);
+        return false;
+    }
+
+    QString scriptCode = "";
```

Comment:
```suggestion
    QString scriptCode;
```

Reply:
done in dad4cc361ff005824b9ea8ff22e6b544e6675283